### DISCUSSION TITLE
Whitelist and Blacklists

### DISCUSSION THREAD

**Discussion by RebbePod** [MrMissx/Telegram_Forwarder#27] [Ideas]
Is it possible to make a whitelist or blacklist?

Also is it possible to tweak the bot to work from any channel without needing to be admin?

**Comment by MrMissx** [comment]
What kind of blacklist/whitelist did you mean?
You couldn't do that because of how the bot API works. It need to be on the channel (of course as admin) to receive an update. you can do that with MTProto (libraries like Telethon / Pyrogram) with userbot by only joining the channel not Bot (PTB).

**Reply by RebbePod** [reply]
A whitelist to only forward messages that contain ...

Gotcha, that's really annoying. I need a simple forward from one channel to the next, but I'm not familiar with Telethon unfortunately.

**Reply by MrMissx** [reply]
You can do it with some tweaks to the code checking if the message recieved contains any words.

There is a project i know that forward **all message at once** with Telethon. Real-time forwarding of incoming messages **(still [WIP](https://github.com/aahnik/tgcf/discussions/43))**.
[aahnik/tgcf](https://github.com/aahnik/tgcf).

**Reply by RebbePod** [reply]
That project looks amazing, thanks!
